plugins/plugin-chart-echarts/src/Gauge/transformProps.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  QueryFormMetric,
+  ChartProps,
+  CategoricalColorNamespace,
+  CategoricalColorScale,
+  DataRecord,
+  getNumberFormatter,
+  getMetricLabel,
+} from '@superset-ui/core';
+import { EChartsOption, GaugeSeriesOption } from 'echarts';
+import { GaugeDataItemOption } from 'echarts/types/src/chart/gauge/GaugeSeries';
+import { parseNumbersList } from '../utils/controls';
+import { DEFAULT_FORM_DATA as DEFAULT_GAUGE_FORM_DATA, EchartsGaugeFormData } from './types';
+
+// https://stackoverflow.com/a/10284006/2858258
+const zip = (rows: any[][]) => rows[0].map((_, c) => rows.map(row => row[c]));
```

Comment:
I think this is a little difficult to read, and I'm not sure it's necessarily needed here (see below)